pkg/proxy/ipvs/proxier.go

```diff
@@ -634,6 +631,13 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 				bmods = append(bmods, module)
 			}
 		}
+
+		for _, nfctmodule := range nfConntrackModules {
```

Comment:
Looks like you have similar for here: https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L634-L638

The only difference is the break which you can avoid and create a single helper function.

Reply:
Actually they are different, because one deals with the conntrack modules and the built-in modules of the kernel, and the other is related to /proc/modules file (when the module is not built in).

I'll take a broader look at the file and see anyway what can be improved based on this

Reply:
I agree that some refactoring would be good here. (It would also be good to move all of the kernel stuff out into a separate file first as a separate commit, and then fix things up in a second commit.) (It's also weird that this function is called `GetModules` but it actually tries to _load_ modules.)

Reply:
@danwinship sorry for the long delay. I'm going to move the module logics to the utils/ipvs package (seems more reasonable) on a first commit, and improve the logics here on other commit.

Maybe everything can be on the utils, with a loadModule function that can deal with that.